cnudie/util.py

```diff
@@ -159,6 +200,23 @@ def find_changed_component(
     )
 
 
+def _enumerate_group_pairs(
+    left_artifacts: typing.List[typing.Union[cm.Resource, cm.ComponentSource, cm.Label]],
+    right_artifacts: typing.List[typing.Union[cm.Resource, cm.ComponentSource, cm.Label]],
+) -> typing.Tuple[typing.List[cm.Resource], typing.List[cm.Resource]]:
+    # group the resources with the same name on both sides
+    for artifact in left_artifacts:
+        right_resource_group = [r for r in right_artifacts if r.name == artifact.name]
+        # get resources for one resource via name
```

Comment:
just to check I understand the semantics properly:

- we accept two sequences of name-bearing objects ("left" and "right")
- both sequences may contain elements w/ the same `name`
- not all names necessarily have to exist in both sequences
- if a name does not exist in the right group (but does in the left), it is dropped from the result-generator
- otherwise, a two-tuple of equally-named elements on left and right is yielded

Reply:
-> is this asymmetry intended? one might also consider to drop elements that do not exist on both sides, as opposed to only dropping those that only exist on the left side